### DISCUSSION TITLE
Ubisys S1 and S2 don't automatically update their state in zigbee2mqtt

### DISCUSSION THREAD

**Discussion by Sailing74** [Koenkk/zigbee2mqtt#19028] [Devices]
I have a number of Ubisys devices and generally am really happy with them. With a S1 and S2, I have a bit of a problem. I can control them from zigbee2mqtt, and their energy etc. values update. But when I switch the device with the attached button (toggle mode), I am not seeing messages on status updates (on/off) in the log, and, thus, zigbee2mqtt doesn't know the correct status. I can manually refresh the status in the web interface, that works.

How can I get the S1/S2 to send automatic messages about their state (on/off) when buttons are pressed? I tried setting state_action to "on", but that didn't change anything. Also, even if I configure state_action as on, the "Exposes" page in z2m webif shows "N/A" under "ACTION" until I manually refresh the status in the z2m webif. When I push the button connected to either device, I am not seeing entries in the log. So it appears the devices are not sending messages when switched.

Did I happen to misconfigure something? This is the S1:

`  '0x001fee0000004aba':
    friendly_name: Lt_Wohnzimmer2
    state_action: false`

`{"id":27,"type":"Router","ieeeAddr":"0x001fee0000004aba","nwkAddr":14297,"manufId":4338,"manufName":"ubisys","powerSource":"Mains (single phase)","modelId":"S1 (5501)","epList":[1,2,3,200,232,242],"endpoints":{
   "1":{
      "profId":260,"epId":1,"devId":266,"inClusterList":[0,3,4,5,6],"outClusterList":[],"clusters":{
         "genBasic":{
            "attributes":{
               "modelId":"S1 5501)","manufacturerName":"ubisys","powerSource":1,"zclVersion":8,"appVersion":1,"stackVersion":1,"hwVersion":6,"dateCode":"20190612-DE-FB1","swBuildId":"1.9.2"}
            },"genOnOff":{
               "attributes":{
               "onOff":0}
            }
         },"binds":[],"configuredReportings":[],"meta":{}
      },"2":{
         "profId":260,"epId":2,"devId":1,"inClusterList":[0,3],"outClusterList":[5,6,8],"clusters":{
            "genBasic":{
               "attributes":{}
            }
         },"binds":[{
            "cluster":6,"type":"endpoint","deviceIeeeAddress":"0x001fee0000004aba","endpointID":1
         }],"configuredReportings":[],"meta":{}
      },"3":{
         "profId":260,"epId":3,"devId":1281,"inClusterList":[0,1794,2820],"outClusterList":[],"clusters":{
            "genBasic":{
               "attributes":{}
            },"seMetering":{
               "attributes":{
                  "multiplier":1,"divisor":1000,"instantaneousDemand":0,"currentSummDelivered":[0,11]
               }
            }
         },"binds":[{
            "cluster":1794,"type":"endpoint","deviceIeeeAddress":"0x00124b002a2e83a1","endpointID":1
         }],"configuredReportings":[{
            "cluster":1794,"attrId":1024,"minRepIntval":5,"maxRepIntval":3600,"repChange":1
         }],"meta":{}
      },"200":{
         "profId":49155,"epId":200,"devId":0,"inClusterList":[],"outClusterList":[],"clusters":{},"binds":[],"configuredReportings":[],"meta":{}
      },"232":{
         "profId":260,"epId":232,"devId":1287,"inClusterList":[0,21,64512],"outClusterList":[3,25],"clusters":{
            "genBasic":{
               "attributes":{}
            },"manuSpecificUbisysDeviceSetup":{
               "attributes":{
                  "inputConfigurations":[0],"inputActions":[{
                     "type":"Buffer","data":[0,13,2,6,0,2]
                  }]
               }
            }
         },"binds":[],"configuredReportings":[],"meta":{}
      },"242":{
         "profId":41440,"epId":242,"devId":102,"inClusterList":[33],"outClusterList":[33],"clusters":{},"binds":[],"configuredReportings":[],"meta":{}
      }
   },"appVersion":1,"stackVersion":1,"hwVersion":6,"dateCode":"20190612-DE-FB1","swBuildId":"1.9.2","zclVersion":8,"interviewCompleted":true,"meta":{
      "configured":3
   },"lastSeen":1695188749881,"defaultSendRequestWhen":"immediate"
}`

Edit: Sorry, I tried to structure the config properly when creating this entry by manually adding indents. But this is not showing in my post. It's still there when I edit it, though.

**Comment by Sailing74** [comment]
@sjorge - I hope it's ok to tag you here. I know you have a number of Ubisys devices running, and hope that you might have a view on what I might be missing/doing wrong here, so that I don't get an updated status in zigbee2mqtt if S1 or S2 are switched using a connected button?

Thanks!

**Reply by sjorge** [reply]
There are a few things that could be at play:
- if your input_actions are bound to the local relay on the S1/S2, you might need to set up reporting for genOnOff.OnOff for the device
- if your input_actions controls a group actions show up in the 'action' attribibute

**Reply by Sailing74** [reply]
Thanks, after googling around, I'm still not entirely sure what I did, but clicking on "Bind" for the endpoint in the "Bind" tab of the z2m WebUI created an additional binding for the coordinator, which apparently solved the reporting:

![image](https://github.com/Koenkk/zigbee2mqtt/assets/5119859/c04c9a5d-13f1-45ec-bac2-a6366e08fb55)

So... Thank you, once again!
